### DISCUSSION TITLE
Sim2real transfer for betaflight

### DISCUSSION THREAD

**Discussion by saghavart** [utiasDSL/gym-pybullet-drones#217] [Q&A]
Hi. I am planning to use your framework for my Reinforcement Learning task. I saw that there is Betaflight support. My question is how can I achieve sim2real transfer for betaflight. Shall I write a new flight mode and build the new firmware? And also betaflight is using his own controller, so how is my RL code integrated on top of it?

**Comment by JacopoPan** [comment]
The simulator currently only supports SITL with Betaflight, to bring it to a real drone, you need to implement your companion board controller on top of the MSP protocol, see https://github.com/betaflight/betaflight/discussions/12615
